modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/Checkbox/Checkbox.es.js

```diff
@@ -261,4 +262,145 @@ describe('Field Checkbox', () => {
+		});
+
+		it('Activating predefinedValue', () => {
+			const {queryByLabelText} = render(
+				<CheckboxWithProvider
+					label="Boolean"
+					predefinedValue={['false']}
+				/>
+			);
+
+			const input = queryByLabelText('Boolean');
+			fireEvent.change(input, {target: {predefinedValue: true}});
+			expect(input.predefinedValue).toBe(true);
+		});
+
+		it('Deactivating predefinedValue', () => {
+			const {queryByLabelText} = render(
+				<CheckboxWithProvider
+					label="Boolean"
+					predefinedValue={['true']}
+				/>
+			);
+
+			const input = queryByLabelText('Boolean');
+			fireEvent.change(input, {target: {predefinedValue: false}});
+			expect(input.predefinedValue).toBe(false);
+		});
+
+		it('Activating boolean field', () => {
+			const {queryByLabelText} = render(
+				<CheckboxWithProvider
+					label="Boolean"
+					predefinedValue={['false']}
+				/>
+			);
+
+			const input = queryByLabelText('Boolean');
+			fireEvent.change(input, {target: {checked: true}});
+			expect(input).toBeChecked();
+		});
+
+		it('Deactivating boolean field', () => {
+			const {queryByLabelText} = render(
+				<CheckboxWithProvider
+					label="Boolean"
+					predefinedValue={['true']}
+				/>
+			);
+
+			const input = queryByLabelText('Boolean');
+			fireEvent.change(input, {target: {checked: false}});
+			expect(input).not.toBeChecked();
+		});
+
+		it('Testing if it shows the label', () => {
+			const {queryByLabelText} = render(
+				<CheckboxWithProvider
+					label="Boolean"
+					predefinedValue={['true']}
+					showLabel
+				/>
+			);
+
+			const input = queryByLabelText('Boolean');
+			expect(input).not.toBeNull();
+		});
+
+		it('Testing if the required icon appears when the field is required', () => {
+			render(<CheckboxWithProvider required={true} />);
+
+			const requiredIcon = document
+				.getElementsByClassName('lexicon-icon-asterisk')
+				.item(0);
+
+			expect(requiredIcon).not.toBeNull();
+		});
+
+		it('Testing if the required icon do not appears when the field is not required', () => {
+			render(<CheckboxWithProvider required={false} />);
+
+			const requiredIcon = document
+				.getElementsByClassName('lexicon-icon-asterisk')
+				.item(0);
+
+			expect(requiredIcon).toBeNull();
+		});
+
+		it('Enabling the switcher in boolean field', () => {
+			const {queryByLabelText} = render(
+				<CheckboxWithProvider
+					label="Boolean"
+					showAsSwitcher={false}
+					showLabel
+				/>
+			);
+
+			const input = queryByLabelText('Boolean');
+			fireEvent.change(input, {target: {showAsSwitcher: true}});
+
+			expect(input.showAsSwitcher).toBe(true);
+		});
+
+		it('Disabling the switcher in boolean field', () => {
+			const {queryByLabelText} = render(
+				<CheckboxWithProvider
+					label="Boolean"
+					showAsSwitcher={true}
+					showLabel
+				/>
+			);
+
+			const input = queryByLabelText('Boolean');
+			fireEvent.change(input, {target: {showAsSwitcher: false}});
+
+			expect(input.showAsSwitcher).toBe(false);
```

Comment:
Prefer to use [DOM Assertions](https://github.com/testing-library/jest-dom) to give more descriptive feedback when test fails.
In this case you can use [toHaveAttribute()](https://github.com/testing-library/jest-dom#tohaveattribute()):
```suggestion
			expect(input).toHaveAttribute('showAsSwitcher', false);
```